src/shared/creds-util.c

```diff
@@ -668,6 +671,12 @@ int encrypt_credential_and_warn(
                         log_debug("Running in container, not attempting to use TPM2.");
 
                 try_tpm2 = r <= 0;
+
+                support = tpm2_support();
+                if (support != TPM2_SUPPORT_FULL) {
+                        log_debug("System lacks TPM2 support, not attempting to use TPM2.");
+                        try_tpm2 = false;
+                }
         } else if (sd_id128_equal(with_key, _CRED_AUTO_INITRD)) {
                 /* If automatic mode for initrds is selected, we'll use the TPM2 key if the firmware does it,
                  * otherwise we'll use a fixed key */
```

Comment:
so, I think we should merge the `_CRED_AUTO_INITRD` and `_CRED_AUTO` branches here, I fail to see why we should treat them differently at this point: here, if a tpm is available we should use it.

the only difference between the two is that in case of _CRED_AUTO_INITRD we should use an empty key if tpm is not available (and no key in /var), while in case of _CRED_AUTO we shouldn't use the empty key but instead the key in /var. But both of this is handled further down, so up here the two cases case be treated the same.

Reply:
hmm, a log message further down says "Firmware reported a TPM2 being present and used, …". This will be misleading once we do the full tpm2_support() check, because then we check much more than the firmware. hence please also update that string then.

Reply:
Should be done